background_worker/jobs/clear_blocks.go

```diff
@@ -10,29 +10,56 @@ import (
 	_ "github.com/lib/pq"
 )
 
+const (
+	datenumHourlyParsing = "2006010215"
+)
+
 type ClearRecrodsParams struct {
 	dbconn.DBConnectionParams
 	RecordRetentionDays int
 }
 
-func ClearBlocks(params ClearRecrodsParams) error {
+type ClearJob struct {
+	now func() time.Time
+}
+
+func WithNow(nowFunc func() time.Time) func(*ClearJob) {
+	return func(p *ClearJob) {
+		p.now = nowFunc
+	}
+}
+
+func NewClearJob(opts ...func(job *ClearJob)) *ClearJob {
+	c := &ClearJob{
+		now: time.Now,
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c
+}
+
+func (c ClearJob) ClearBlocks(params ClearRecrodsParams) error {
 	Log(INFO, "Connecting to database ...")
 
 	conn, err := sqlx.Open(params.Scheme(), params.String())
 	if err != nil {
 		Log(ERROR, "unable to create connection")
 		return err
 	}
-	interval := fmt.Sprintf("%d days", params.RecordRetentionDays)
 
-	stmt, err := conn.Preparex("DELETE FROM blocks WHERE inserted_at <= (CURRENT_DATE - $1::interval)")
+	start := c.now()
+	datenumRetentiondDays := start.Add(-24 * time.Hour * time.Duration(params.RecordRetentionDays))
+
+	stmt, err := conn.Preparex("DELETE FROM blocks WHERE inserted_at_num <= $1::int")
```

Comment:
what is inserted_at_num and how does it differ from inserted_at? Why we can't use inserted_at?